### DISCUSSION TITLE
How to intercept Bevy's AppExit event for long enough to do necessary cleanup ops

### DISCUSSION THREAD

**Discussion by crabdancing** [bevyengine/bevy#17002] [Q&A]
I experimented a bit, but I wasn't able to figure out how to do this with the basic documentation. There are several things I might want to do, including writing files to disk before exit. The problem is, the Bevy window seems to just send `AppExit` directly and doesn't give me a chance to activate systems in response?

Edit: I am aware that this question was asked before (in 2022), but the answer is no longer applicable as of Bevy 0.15.

**Comment by NiklasEi** [comment]
You can configure the `WindowPlugin` to not send an `AppExit` event when all or the primary window are closed via the `exit_condition` parameter. You could then listed to window closed events your self, run all your shut down tasks and finally fire an `AppExit` event.
